File: StringCalculator.js
class StringCalculator {
  add(numbers) {
    if (!numbers) return 0

    let delimiter = /,|\n/
    if (numbers.startsWith('//')) {
      const delimiterMatch = numbers.match(/^\/\/(.+)\n/)
      delimiter = new RegExp(delimiterMatch[1])
      numbers = numbers.slice(delimiterMatch[0].length)
    }

    const nums = numbers
      .split(delimiter)
      .map((num) => parseInt(num, 10))
      .filter((num) => !isNaN(num))

    const negatives = nums.filter((num) => num < 0)
    if (negatives.length) {
      throw new Error(
        `Negative Numbers are not allowed! : ${negatives.join(', ')}`
      )
    }
  }
}

// Exporting the class for unit testing
module.exports = StringCalculator

File: string_calculator.py
class StringCalculator:
    def add(self, numbers: str) -> int:
        if not numbers:
            return 0

        delimiter = ",|\n"
        if numbers.startswith("//"):
            parts = numbers.split("\n", 1)
            custom_delimiter = parts[0][2:]
            delimiter = custom_delimiter
            numbers = parts[1]

        import re
        nums = re.split(delimiter, numbers)

        nums = [int(num) for num in nums if num.isdigit()]

        negatives = [num for num in nums if num < 0]
        if negatives:
            raise ValueError(f"Negative numbers not allowed: {', '.join(map(str, negatives))}")

        return sum(nums)
